app/src/main/java/mozilla/lockbox/action/SettingAction.kt

```diff
@@ -44,6 +46,24 @@ class Setting {
                     Never -> R.string.never
                 }
             }
+
+        val seconds: Int by lazy {
+            // I'm concerned about this method of setting time. I wonder if it's going
+            // to be circumventable by changing the device time. we might want to consider a
+            // options for mitigating that attack vector.
+            when (this) {
+                OneMinute -> 60
+                FiveMinutes -> 60 * 5
+                FifteenMinutes -> 60 * 15
+                ThirtyMinutes -> 60 * 30
+                OneHour -> 60 * 60
+                TwelveHours -> 60 * 60 * 12
+                TwentyFourHours -> 60 * 60 * 24
+                Never -> 0
+            }
+        }
+
+        val ms: Long by lazy { (this.seconds * 1000).toLong() }
```

Comment:
Enums in kotlin can have their own constructors and properties.

You can get rid of your Swift-y switches :); for example:

e.g.
```kotlin
enum AutoLockTime(
  val stringValue: @StringRes Int,
  private val minutes: Long
) {
    val second: Long = minutes * 60L
    val ms: Long = seconds * 1000L

    OneMinute(R.string.one_minute, 1),
    TwoMinute(R.string.one_minute, 2),
    …
}
```

Reply:
oooo